xwiki-platform-core/xwiki-platform-refactoring/xwiki-platform-refactoring-api/src/main/java/org/xwiki/refactoring/job/DeleteRequest.java

```diff
@@ -141,6 +147,35 @@ public void setUpdateLinks(boolean updateLinks)
         setProperty(UPDATE_LINKS, updateLinks);
     }
 
+    /**
+     * @return {@code true} if, when {@link #isUpdateLinks()} is {@code true}, the refactoring shall wait for link
+     * indexing to complete. This ensures that accurate information about links is available, this is particularly
+     * relevant when multiple documents with links between them are moved.
+     *
+     * @since 16.8.0
+     * @since 16.4.4
+     * @since 15.10.13
+     */
+    @Unstable
+    public boolean isWaitForIndexing()
```

Comment:
Shouldn't MoveAttachmentRequest have a true by default `#isWaitForIndexing()` too ? (since we refactor links to the moved attachments AFAIK, in MovedAttachmentListener, use case which seems to be missing in your pull request right now).

I think a EntityRequest#isWaitForIndexing (false by default) would make sense (DeleteRequest, AbstractCopyOrMoveRequest and MoveAttachmentRequest would just overwrite the default value to be true instead of false). Would make

Reply:
I've moved the methods to `EntityRequest` but I set them to `true` by default as the waiting is only actually happening when links are refactored. Therefore, I think it is okay to have it `true` by default. Also, I've added the waiting logic to `MovedAttachmentListener`, refactoring the code to avoid duplicating the logic for it.